code/modules/medical/patientprocessor.dm

```diff
@@ -0,0 +1,21 @@
+/obj/machinery/medical/patient_processor
+	name = "patient processor"
+	desc = "A device that uses advanced AutoDoc technology to heal patients."
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "unloader"
+	density = 1
+	anchored = 1
+	var/input_dir = 1
+	var/output_dir = 2
+
+/obj/machinery/medical/patient_processor/process()
+	if((!output_dir || !input_dir) || output_dir == input_dir)
```

Comment:
instead of just checking for null/0, you should confirm that input and output are valid dirs, e.g.:

`!(output_dir in cardinal)`